feat: Add migration guide for Camunda Process Test (#6326)
camunda/camunda-docs: docs/apis-tools/migration-manuals/migrate-to-camunda-process-test.md

you can't install a Docker-API compatible container runtime, such as Docker on Linux or Docker Desktop.

Prepare your remote runtime:

1. Install [Camunda 8 Run](/self-managed/quickstart/developer-quickstart/c8run.md#install-and-start-camunda-8-run) on
   your machine
2. [Enable the management clock endpoint](/apis-tools/testing/configuration.md#prerequisites-1)

- Create an `application.yaml` file in the root `/c8run` directory
- Add the property `zeebe.clock.controlled: true`

3. Start Camunda 8 Run

Then, switch CPT's runtime mode to `remote` in your project.

<Tabs groupId="client" defaultValue="spring-sdk" queryString values={
[
{label: 'Camunda Spring Boot SDK', value: 'spring-sdk' },
{label: 'Java client', value: 'java-client' }
]
}>

<TabItem value='spring-sdk'>

In your `application.yml` (or `application.properties`):

```yaml
io:
  camunda:
    process:
      test:
        runtime-mode: remote
```

</TabItem>

<TabItem value='java-client'>

In your process test class, register the JUnit extension in the following way:

```java
package com.example;

import io.camunda.process.test.api.CamundaProcessTestExtension;

> import org.junit.jupiter.api.extension.RegisterExtension;

Author: ```suggestion
import io.camunda.process.test.api.CamundaProcessTestRuntimeMode;
import org.junit.jupiter.api.extension.RegisterExtension;
```

---
id: migrate-to-camunda-process-test
title: Migrate to Camunda Process Test
description: "Learn how to migrate from Zeebe Process Test to Camunda Process Test"
---

import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";

> [Camunda Process Test](/apis-tools/testing/getting-started.md) (CPT) is the successor to [Zeebe Process Test](/apis-tools/java-client/zeebe-process-test.md) (ZPT).  

Reviewer: ```suggestion
[Camunda Process Test](/apis-tools/testing/getting-started.md) (CPT) is a library to test your BPMN processes and your process applications. It is the successor to [Zeebe Process Test](/apis-tools/java-client/zeebe-process-test.md) (ZPT).  
```

> Since version **8.8**, CPT is available and fully supported. ZPT is deprecated and will be removed in version **8.10**.  
> See the [announcement](https://camunda.com/blog/2025/04/camunda-process-test-the-next-generation-testing-library/) for details.

Reviewer: ```suggestion
Starting with version **8.8**, ZPT is deprecated and will be removed in version **8.10**.  
See the [announcement](https://camunda.com/blog/2025/04/camunda-process-test-the-next-generation-testing-library/) for details.
```

This guide walks you through migrating your existing test cases from ZPT to CPT step-by-step.

:::note
Be aware that there are differences between ZPT and CPT in both API and behavior, which may increase migration effort depending on your existing test cases.

**Key differences:**

> - **Underlying engine:** ZPT uses only Camunda’s workflow engine (Zeebe) with access to internal components, whereas CPT runs the full Camunda distribution and interacts with the public API.

Reviewer: ```suggestion
- **Underlying engine:** ZPT uses only Camunda’s workflow engine (Zeebe) with access to internal components, whereas CPT runs the full Camunda distribution and interacts with the Orchestration Cluster API.
```

- **Assertions and utilities:** CPT uses different names to align with the API, and not all ZPT assertions/utilities have equivalents in CPT.

> - **Performance:** CPT is slower because it runs the full distribution.

Reviewer: Can we change this sentence to be more positive?  Something like
```suggestion
- **Startup time:** CPT takes longer to start as it runs the full Orchestration Cluster distribution.
```

**Key advantages of CPT:**

> - Access to Camunda’s full API

Reviewer: ```suggestion
- Access to Camunda’s Orchestration Cluster API and Connectors
```

- Support for Camunda user tasks
- Blocking assertions for asynchronous processing
- Enhanced mocking utilities
  :::

> 

Reviewer: ```suggestion
## How to migrate to Camunda Process Test from Zeebe Process Test?
```

Reviewer 2: I don't think that adds value, given the page is already the migration guide itself called "Migrate to Camunda Process Test". Would you be fine not introducing this dedicated section?

> ## Update your dependency

Reviewer: ```suggestion
### Update your dependency
```

First, update your Maven dependency.

- **If you use ZPT with Camunda Spring Boot SDK integration**  
  (`artifactId: spring-boot-starter-camunda-test` or `spring-boot-starter-camunda-test-testcontainer`),  
  replace it with **CPT’s Spring integration module**.

- **If you use ZPT without Spring**  
  (`artifactId: zeebe-process-test-extension` or `zeebe-process-test-extension-testcontainer`),  
  replace it with **CPT’s Java module**.

<Tabs groupId="client" defaultValue="spring-sdk" queryString values={
[
{label: 'Camunda Spring Boot SDK', value: 'spring-sdk' },
{label: 'Java client', value: 'java-client' }
]
}>

<TabItem value='spring-sdk'>

In your Maven `pom.xml`, add the dependency:

```xml
<dependency>
  <groupId>io.camunda</groupId>
  <artifactId>camunda-process-test-spring</artifactId>
  <scope>test</scope>
</dependency>
```

</TabItem>

<TabItem value='java-client'>

In your Maven `pom.xml`, add the dependency:

```xml
<dependency>
  <groupId>io.camunda</groupId>
  <artifactId>camunda-process-test-java</artifactId>
  <scope>test</scope>
</dependency>
```

</TabItem>
</Tabs>

> ## Choose your runtime

Reviewer: ```suggestion
### Choose your runtime
```

> Next, choose your CPT runtime depending on your environment.

Reviewer: ```suggestion
Next, choose how you want to run CPT, considering your environment. CPT can be used in two modes: 

- CPT with Testcontainers (as equivalent to ZPT with Testcontainers)
- CPT with remote engine (as equivalent to ZPT's embedded runtime)
```

> ### ZPT with Testcontainers

Reviewer: ```suggestion
#### ZPT with Testcontainers
```

Reviewer 2: see https://github.com/camunda/camunda-docs/pull/6326#discussion_r2293888415

If you use ZPT with Testcontainers (

> `artifactId: zeebe-process-test-extension-testcontainer/spring-boot-starter-camunda-test-testcontainer`), then you can
> use CPT's default [Testcontainers runtime](/apis-tools/testing/configuration.md#testcontainers-runtime) without
> additional changes.

Reviewer: ```suggestion
`artifactId: zeebe-process-test-extension-testcontainer/spring-boot-starter-camunda-test-testcontainer`), then you can
use CPT's default Testcontainers runtime without
additional changes. Check the documentation on how to use CPT with Testcontainers [here](/apis-tools/testing/configuration.md#testcontainers-runtime)
```

> ### ZPT's embedded runtime

Reviewer: ```suggestion
#### ZPT's embedded runtime
```

Reviewer 2: see https://github.com/camunda/camunda-docs/pull/6326#discussion_r2293888415

If you use ZPT’s **embedded runtime**  
(`artifactId: zeebe-process-test-extension` or `spring-boot-starter-camunda-test`),  
switch to CPT’s [remote runtime](/apis-tools/testing/configuration.md#remote-runtime).  
Choose this option only if you cannot install a Docker-API compatible container runtime (e.g., Docker on Linux or Docker Desktop).

> 

Reviewer: ```suggestion

In this mode, CPT connects to a remote runtime, such as a local Camunda 8 Run running on your machine.
```
